src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -255,31 +258,35 @@ internal TabItem AddTabItem(IViewExtension viewExtension, ContentControl content
         // This method triggers the close operation on the selected tab. 
         private void CloseTab(object sender, RoutedEventArgs e)
         {
-            string tabName = (sender as Button).CommandParameter.ToString();
-            CloseTab(tabName);
+            string tabName = (sender as Button).DataContext.ToString();
```

Comment:
when the button is constructed you can actually use the `tag` property to add the name of the tab instead of getting it this way - I think that makes it more explicit.

Reply:
hmm, maybe this is not possible - am I understanding correctly the tabitems close buttons are already present from the existing control?

Reply:
They are present in the dynamoview.xaml and all the buttons have the same name. 
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Views/Core/DynamoView.xaml#L1721.

Reply:
theres probably some way to tag them with the header property of their dataContext instead of the name which was never set in the tabItem constructor.